Documentation/provisioning.md

```diff
@@ -0,0 +1,229 @@
+Provisioning a Device
+==============
+
+This document describes the first boot for a device being provisioned in a factory along with the firmware customizations required to enable the flow.
+
+Assumptions:
+* The SD Card or eMMC already contains the OS/Firmware image.
+* The device has an eMMC with RPMB to support secure storage for fTPM and Authenticated Variables.
+* The device being provisioned has an i.MX6 Quad SoC. Other SoCs will require fusing customizations.
+
+Missing features required for a complete provisioning implementation:
+* A connection to an OEM database to assign per-device information such as MAC addresses, serial numbers, and other SMBIOS customizations.
+* A connection to an OEM database to save the TPM Endorsement Key Certificates.
+* Host side cross-signing of the Endorsement Key Certificate to prove that the TPM is trusted by the OEM.
+* Factory integration to power off the device and start provisioning the next on success or to raise an alert on failure.
+* Saving Secure Boot UEFI variables to the device to enable secure boot. The required boot critical drivers are not signed yet.
+* fTPM Endorsement Primary Seed derived from secret unique identity retrieved from a Trusted/Secure CAAM after High Assurance Boot. Currently the secret is randomly generated and will be different if RPMB is cleared.
+
+## In-factory First Boot Flow
+* U-Boot SPL (SRKH and MAC address Provisioning)
+  1) U-Boot SPL boots into *spl_board_provision( )* which does the following:
+  1) Senses that the system is not in Secure or Trusted state (aka not High Assurance Boot).
+  2) Retrieves the SKRH from SPL's device tree and fuses it onto the SoC.
+  3) Requests a MAC address from the provisioning host and fuses it onto the SoC (if MAC not already fused).
+  4) SPL resets the platform.
+  5) After the reset *spl_board_provision( )* senses that the system is in Trusted state so boot continues.
+* Early UEFI (fTPM Initialization)
+  1) The system boots all the way through to UEFI.
+  2) UEFI starts the fTPM TA and then the fTPM attempts to access the eMMC RPMB.
```

Comment:
Unless you driver has caused the fTPM to be enabled earlier, the authvars TA will be the first to run. Maybe leave this ambiguous (ie UEFI starts the secure firmware TAs, which then access the eMMC...)?